tools/scripts/clang_tidy_check.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# This script checks code files with clang-tidy
+# Example usage: ./tools/scripts/check_clang_tidy.sh hook|ci_pull_request_ci_full
+
+set -e
+
+MODE=$1
+
+FILE_FILTER="\.(h|hpp|inl|c|cpp)$"
+FILE_BLACKLIST='(test|testing|tools|iceoryx_posh|iceoryx_dds|iceoryx_binding_c|iceoryx_dds|doc|iceoryx_integrationtest|iceoryx_meta|iceoryx_examples)'
+
+hash run-clang-tidy-12.py || fail "run-clang-tidy-12 not found, please install clang-tidy-12"
+
+fail() {
+    printf "\033[1;31merror: %s: %s\033[0m\n" ${FUNCNAME[1]} "${1:-"Unknown error"}"
+    exit 1
+}
+
+WORKSPACE=$(git rev-parse --show-toplevel)
+cd "${WORKSPACE}"
+
+if ! [[ -f build/compile_commands.json ]]; then # or we just exit here with an error
+    sudo apt-get update && sudo apt-get install -y libacl1-dev libncurses5-dev bison clang-tidy-12
+    export CXX=clang++-12
```

Comment:
Again ubuntu/specific. Could we please use again the trick drafted above to find the correct command and version. And select either `clang++-12` or `clang++` if it exists. Same for `clang-12`.